app/javascript/packages/document-capture/components/in-person-full-address-search.tsx

```diff
@@ -141,17 +144,28 @@ function FullAddressSearch({
     validatedZipCodeFieldRef,
   } = useUspsLocations(locationsURL);
 
-  const inputChangeHandler =
+  const inputChangeHandlerForZipCode =
     <T extends HTMLElement & { value: string }>(input) =>
     (event: React.ChangeEvent<T>) => {
       const { target } = event;
-      input(target.value);
+      const scrubbedZip = target.value.replace(/[^0-9]/g, '');
+      if (scrubbedZip.length > 5) {
+        const zipcodePieces: string[] = [];
+        zipcodePieces.push(scrubbedZip.substring(0, 5));
+        zipcodePieces.push(scrubbedZip.substring(5));
+        input(zipcodePieces.join('-'));
+      } else {
+        input(scrubbedZip);
+      }
     };
 
-  const onAddressChange = inputChangeHandler(setAddressValue);
-  const onCityChange = inputChangeHandler(setCityValue);
-  const onStateChange = inputChangeHandler(setStateValue);
-  const onZipCodeChange = inputChangeHandler(setZipCodeValue);
+  type InputChangeEvent = React.ChangeEvent<HTMLInputElement>;
+  const onAddressChange = (e: InputChangeEvent) =>
+    setAddressValue(e.target.value.replace(/[^a-zA-Z0-9_ ]/gi, '').trimStart());
+  const onCityChange = (e: InputChangeEvent) =>
+    setCityValue(e.target.value.replace(/[^a-zA-Z0-9-'_ ]/gi, '').trimStart());
+  const onStateChange = (e: InputChangeEvent) => setStateValue(e.target.value.trimStart());
+  const onZipCodeChange = inputChangeHandlerForZipCode(setZipCodeValue);
```

Comment:
I think it would be ideal to make the behavior on this page match closely the behavior on other address pages. The differences I noticed are:

1. You can enter leading spaces on other pages. We could allow that and instead just trim leading and trailing spaces before making the API request, but allow them to be entered into the form inputs
2. Address and city special character validations are different -- we allow some additional special characters on other address pages. The characters allowed in the existing page are defined [here](https://github.com/18F/identity-idp/blob/1e7362d13ebbe8ebdf727547c3ac0efa8d1c6e16/app/validators/idv/form_state_id_validator.rb#L14-L32)

Reply:
Note that transliteration takes place _before_ the `reject_chars` regex gets evaluated on each character in the string for the example you mentioned.

The transliteration is based on the I18n library defaults in addition to the list in `config/locales/transliterate/en.yml`. The full list of characters is documented [here](https://docs.google.com/document/d/1sC4L8rICmhYL-OWlsgw4f53fX155QZcpYaPHJbXCiw8/edit).

Reply:
The regexes on the address and city seem too restrictive to meet the story requirements as stated. There's no opportunity to transliterate if the user's non-Latin characters are removed upfront.

Reply:
If we're not able to take advantage of the existing transliteration protections pretty easily (which I'm not sure about just looking at the code) I would suggest we pull that out into a separate ticket, or at least a separate PR

Reply:
I'm ok with deferring the work, but the ticket creation & update should be done and shared with the team before proceeding with the changes in this PR. This does not meet the ticket's AC as-is.

The additional work to meet the current AC looks to me like it would involve attaching the existing transliterable validator to the PO search endpoint, modifying the response code to propagate the validation errors, and adding transliteration to the backend call-through to USPS.

Reply:
I think the next steps are as follows:

1. If my understanding is correct, I will open a new ticket, edit the AC in this ticket, and inform the team of the changes before continuing with steps 2-5
2. Remove regex for address and city in this PR. We will need to be less strict to allow users to enter other characters that will be transliterated in another ticket/PR
3. Open a new issue to handle transliteration right before we hit the PO search endpoint. Handle errors when characters cannot be transliterated
4. Keep zip code validation as is in the PR
5. keep removal of white space (leading/trailing) as is in this PR

Reply:
I would modify that as follows:

> 1. If my understanding is correct, I will open a new ticket, edit the AC in this ticket, and inform the team of the changes before ~continuing with steps 2-5~proceeding
> 2. Remove regex for address and city in this PR. We will need to be less strict to allow users to enter other characters that will be transliterated in another ticket/PR
> 3. Open a new issue to handle transliteration **and sanitization** right before we hit the PO search endpoint. ~Handle~**Display** errors **to user** when characters cannot be transliterated
> 4. Keep zip code validation as is in the PR
> 5. keep removal of white space (leading/trailing) as is in this PR
> 6. **Display errors to the user for blank address and city fields**

Reply:
The transliteration and further sanitation of Address and City will be handled in [LG-10783](https://cm-jira.usa.gov/browse/LG-10783) Transliterate & Sanitize PO search full address form.